test/gtest/uct/uct_test.h

```diff
@@ -187,17 +188,34 @@ class uct_test : public testing::TestWithParam<const resource*>,
     };
 
     template <typename T>
-    static std::vector<const resource*> filter_resources(const std::vector<T>& resources,
-                                                         const std::string& tl_name)
+    static void filter_by_name(std::vector<const resource*> &result,
+                               const T& res, const std::string& tl_name)
+    {
+        if ((tl_name.empty() || (res.tl_name == tl_name))) {
+            result.push_back(&res);
+        }
+    }
+
+    template <typename T>
+    static void commit_noop(std::vector<const resource*> &result,
+                            T& arg) { }
+
+    template <typename T, typename M> static std::vector<const resource*>
+    filter_resources(const std::vector<T>& resources,
```

Comment:
we don't need to make it generic with filter and commit functions. can you pls simplify the code?

Reply:
ok, sure - the code has been simplified to be clear